lib/src/utils.dart

```diff
@@ -499,6 +503,113 @@ String yamlToString(data) {
   return buffer.toString();
 }
 
+/// [YamlMap] which overrides entries in the [original] map with entries from
+/// the [overrides] map.
+///
+/// This map is useful to apply overrides from one yaml file to another, while
+/// keeping source references intact.
+///
+/// The map contains entries for all of the keys of [original]. If both
+/// [original] and [overrides] contain an entry for a given key, [mergeEntry]
+/// is called to merge the values. Otherwise, the entry from [original] is taken
+/// over as is.
+///
+/// Entries that only exist in [overrides] are taken over as is.
+///
+/// The [YamlNode]s of keys of merged entries always come from [original].
+class OverrideYamlMap with MapMixin, UnmodifiableMapMixin implements YamlMap {
```

Comment:
Can't we just mutate the original `YamlMap` or create a new `YamlMap` with the new values?

Reply:
`YamlMap.nodes` is unmodifiable and `YamlMap` has no public API (`YamlMap.internal` is internal) to create a new one from`YamlNode`s. We need to be able to build the new map from `YamlNodes` from two different `YamlMap`s if we want to keep the source references of the `YamlNode`s intact, for useful error messages.